drivers/video/video_stm32_jpeg.c

```diff
@@ -0,0 +1,589 @@
+	{
+		.pixelformat = VIDEO_PIX_FMT_JPEG,
+		.width_min = 1,
+		.width_max = 65535,
+		.height_min = 1,
+		.height_max = 65535,
+		.width_step = 1,
+		.height_step = 1,
+	},
+	{
+		.pixelformat = VIDEO_PIX_FMT_NV12,
+		.width_min = 16,
+		.width_max = 65520,
+		.height_min = 16,
+		.height_max = 65520,
+		.width_step = 16,
+		.height_step = 16,
+	},
+	{0}
+};
+
+static int stm32_jpeg_get_caps(const struct device *dev, struct video_caps *caps)
+{
+	caps->format_caps = stm32_jpeg_fmts;
+
+	caps->min_vbuf_count = 1;
+	caps->min_line_count = LINE_COUNT_HEIGHT;
+	caps->max_line_count = LINE_COUNT_HEIGHT;
+
+	return 0;
+}
+
+static DEVICE_API(video, stm32_jpeg_driver_api) = {
+	.set_format = stm32_jpeg_set_fmt,
+	.get_format = stm32_jpeg_get_fmt,
+	.set_stream = stm32_jpeg_set_stream,
+	.enqueue = stm32_jpeg_enqueue,
+	.dequeue = stm32_jpeg_dequeue,
+	.get_caps = stm32_jpeg_get_caps,
+};
+
+static int stm32_jpeg_enable_clock(const struct device *dev)
+{
+	const struct stm32_jpeg_config *config = dev->config;
+	const struct device *cc_node = DEVICE_DT_GET(STM32_CLOCK_CONTROL_NODE);
+
+	if (!device_is_ready(cc_node)) {
+		LOG_ERR("clock control device not ready");
+		return -ENODEV;
+	}
+
+	/* Turn on JPEG peripheral clock */
+	return clock_control_on(cc_node, (clock_control_subsys_t)&config->jpeg_hclken);
+}
+
+static int stm32_jpeg_init(const struct device *dev)
+{
+	const struct stm32_jpeg_config *cfg = dev->config;
+	struct stm32_jpeg_data *data = dev->data;
+	HAL_StatusTypeDef hret;
+	int ret;
+
+	data->dev = dev;
+
+	/* Enable DCMIPP / CSI clocks */
+	ret = stm32_jpeg_enable_clock(dev);
+	if (ret < 0) {
+		LOG_ERR("Clock enabling failed.");
+		return ret;
+	}
+
+	/* Reset JPEG */
+	if (!device_is_ready(cfg->reset_jpeg.dev)) {
+		LOG_ERR("reset controller not ready");
+		return -ENODEV;
+	}
+	reset_line_toggle_dt(&cfg->reset_jpeg);
+
+	/* Run IRQ init */
+	cfg->irq_config(dev);
+
+#if defined(CONFIG_SOC_SERIES_STM32N6X)
+	HAL_RIF_RISC_SetSlaveSecureAttributes(RIF_RISC_PERIPH_INDEX_JPEG,
+					      RIF_ATTRIBUTE_PRIV | RIF_ATTRIBUTE_SEC);
+#endif
+
+	/* Initialise default input / output formats */
+	k_mutex_init(&data->lock);
+	k_fifo_init(&data->m2m.in.fifo_in);
+	k_fifo_init(&data->m2m.in.fifo_out);
+	k_fifo_init(&data->m2m.out.fifo_in);
+	k_fifo_init(&data->m2m.out.fifo_out);
+
+	/* Initialize default formats */
+	data->m2m.in.fmt.type = VIDEO_BUF_TYPE_INPUT;
+	data->m2m.in.fmt.width = STM32_JPEG_DEFAULT_WIDTH;
+	data->m2m.in.fmt.height = STM32_JPEG_DEFAULT_HEIGHT;
+	data->m2m.in.fmt.pixelformat = STM32_JPEG_DEFAULT_IN_FMT;
+	data->m2m.out.fmt.type = VIDEO_BUF_TYPE_OUTPUT;
+	data->m2m.out.fmt.width = STM32_JPEG_DEFAULT_WIDTH;
+	data->m2m.out.fmt.height = STM32_JPEG_DEFAULT_HEIGHT;
+	data->m2m.out.fmt.pixelformat = STM32_JPEG_DEFAULT_OUT_FMT;
+
+	ret = video_init_ctrl(&data->jpeg_quality, dev, VIDEO_CID_JPEG_COMPRESSION_QUALITY,
+			      (struct video_ctrl_range) {.min = 5, .max = 100,
+							 .step = 1, .def = 50});
+	if (ret) {
```

Comment:
Nit: for consistency
```suggestion
	if (ret < 0) {
```